File: uniffi_bindgen/src/bindings/swift/templates/Int16Helper.swift
fileprivate struct FfiConverterInt16: FfiConverterPrimitive {
    typealias FfiType = Int16
    typealias SwiftType = Int16

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> Int16 {
        return try lift(readInt(&buf))
    }

    static func write(_ value: Int16, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}

File: uniffi_bindgen/src/bindings/kotlin/templates/Int16Helper.kt
internal object FfiConverterI16 {
    fun lift(v: Short): Short {
        return v
    }

    fun read(buf: ByteBuffer): Short {
        return buf.getShort()
    }

    fun lower(v: Short): Short {
        return v
    }

    fun write(v: Short, buf: RustBufferBuilder) {
        buf.putShort(v)
    }
}
